### ISSUE TITLE
Create Cypress.Commands.overwrite API

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress#471]
Instead of obliterating existing commands, there should be an easy way to overwrite existing ones in order to modify their logic.

Something like this...

```js
Cypress.Commands.overwrite("visit", function(orig, url, options){
  // modify url or options here...
  return orig(url, options)
})
```

**Comment by jennifer-shehane** [comment]
The code for this is done, but this has yet to be released. We'll update this issue and reference the [changelog](https://on.cypress.io/changelog) when it's released.

**Comment by brian-mann** [comment]
Fixed in [`0.20.0`](https://docs.cypress.io/guides/references/changelog.html#0-20-0)
